### PR TITLE
Gherkin feature/workflow writeups for LocationType/Location and Namespace/Prefix

### PR DESCRIPTION
**Pull request by glennmatthews**
# Closes: #4781
# What's Changed

Add Gherkin feature writeups for the following basic user workflows:
- Creating a hierarchy of LocationTypes and Locations
- Creating a Namespace and a hierarchy of Prefixes

No test automation at this time - these are purely for manual testing at present.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- n/a Attached Screenshots, Payload Example
- n/a Unit, Integration Tests
- n/a Documentation Updates (when adding/changing features)
- n/a Example Plugin Updates (when adding/changing features)
- [x] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/tests/features/locations.feature [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      | Geographic Region | None   | True     |
+    And Nautobot shows the "LocationType Detail" view for the LocationType with name "Geographic Region"
+
+    When I click the "Add child" button
+    Then Nautobot shows the "Add LocationType" view with the following presets:
+      | parent            |
+      | Geographic Region |
+    When I make the following form entries:
+      | name |
+      | city |
```
Comment (HanlinMiao) [inline nautobot/dcim/tests/features/locations.feature:30]: Capitalization mismatch with line 34 perhaps?

File: nautobot/dcim/tests/features/locations.feature [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      | name     | parent | nestable | content_types |
+      | Building | City   | False    | [dcim.device] |
+    And Nautobot shows the "LocationType Detail" view for the LocationType with name "Building"
+
+  Scenario: Creating a hierarchy of Locations
+    Given that LocationType records exist with the following properties:
+      | name              | parent            | nestable |
+      | Geographic Region | None              | True     |
+      | City              | Geographic Region | False    |
+      | Building          | Building          | False    |
```
Comment (HanlinMiao) [inline nautobot/dcim/tests/features/locations.feature:55]:
```suggestion
      | name              | parent            | nestable |
      | Geographic Region | None              | True     |
      | City              | Geographic Region | False    |
      | Building          | City              | False    |
```

File: nautobot/dcim/tests/features/locations.feature [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    And I submit the form
+    Then a Location exists in Nautobot with the following properties:
+      | name          | location_type     | parent | status |
+      | United States | Geographic Region | None   | Active |
+    And Nautobot shows the "Location Detail" view for the Location with name "United States"
+
+    When I click the "Add child" button
+    Then Nautobot shows the "Add Location" view with the following presets:
+      | parent        |
+      | United States |
```
Comment (HanlinMiao) [inline nautobot/dcim/tests/features/locations.feature:69]: ~Do we need to add a `location_type` preset here as well or no?~
NVM you still have to pick from the dropdown

File: nautobot/ipam/tests/features/prefixes.feature [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    When I navigate to the "Add Namespace" view
+    And I make the following form entries:
+      | name | description       |
+      | CORP | Corporate network |
+    And I submit the form
+    Then a Namespace exists in Nautobot with the following properties:
+      | name | description       |
+      | CORP | Corporate network |
+    And Nautobot shows the "Namespace Detail" view for the Namespace with name "CORP"
+    # TODO: it would be nice to have the ability to create a Prefix directly from the Namespace detail view?
```
Comment (HanlinMiao) [inline nautobot/ipam/tests/features/prefixes.feature:23]: Agreed

File: nautobot/ipam/tests/features/prefixes.feature [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      | 10.0.96.0/22  | —       | —      |
+      | 10.0.100.0/24 | Network | Active |
+
+    When I click the "10.0.96.0/22" link
+    Then Nautobot shows the "Add Prefix" view with the following presets:
+      | prefix        | namespace |
+      | 10.0.96.0/22  | CORP      |
+    When I make the following form entries:
+      | type         | type      | status |
+      | 10.0.96.0/20 | Container | Active |
```
Comment (HanlinMiao) [inline nautobot/ipam/tests/features/prefixes.feature:85]: Perhaps?
```suggestion
      | parent       | type      | status |
      | 10.0.96.0/20 | Container | Active |
```

File: nautobot/ipam/tests/features/prefixes.feature [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      | prefix      | type      | status |
+      | 10.0.0.0/8  | Container | Active |
+      | 10.0.0.0/16 | Container | Active |
+    And the tab "Child Prefixes" includes a badge with the number "1"
+
+    When I navigate to the "Prefix List" view
+    And specify a filter of "?namespace=CORP"
+    Then the table "Prefixes" includes rows with the following properties:
+      | prefix        | namespace | type      | status | children | utilization |
+      | 10.0.0.0/8    | CORP      | Container | Active | 3        | 0%          |
```
Comment (HanlinMiao) [inline nautobot/ipam/tests/features/prefixes.feature:102]: I am not sure how utilization is calculated here. But with 3 children utilization should be > 0%? Ignore this if I am completely wrong.

File: nautobot/dcim/tests/features/locations.feature (hunk 1 above)
Comment (glennmatthews) [reply nautobot/dcim/tests/features/locations.feature:30]:
```suggestion
      | City |
```

File: nautobot/ipam/tests/features/prefixes.feature (hunk 5 above)
Comment (glennmatthews) [reply nautobot/ipam/tests/features/prefixes.feature:85]:
```suggestion
      | prefix       | type      | status |
      | 10.0.96.0/20 | Container | Active |
```

File: nautobot/ipam/tests/features/prefixes.feature (hunk 6 above)
Comment (glennmatthews) [reply nautobot/ipam/tests/features/prefixes.feature:102]: A /16 allocated out of a /8 is 1/256 utilization which rounds down to 0%. Probably would be good to add a comment here though.

**Review by HanlinMiao — APPROVED** [review]

**Pull request merged**